### PR TITLE
Improvements on Temoa Model

### PR DESCRIPTION
**Pull request by samgdotson**
This pull request includes:
- Fixes for the Snakefile
- Adds an experiment to the ``temoa_processing`` notebook. 
- Adds an ``optimal_sizing`` notebook that
   - Builds all figures by running snakemake
   - Builds a specialized business-as-usual figure (with reference year 2014). 
- Fixes Temoa configuration files
- Fixes Temoa databases (``01_uiuc`` and ``03_uiuc`` were actually the same database they are now different). 
- Updates ``data_parser.py`` by improving 
   - figure fonts, 
   - sizes, 
   - simplifies titles, 
   - and adds flexibility with notebooks (by allowing users to specify if they want to save or show the figure). 
- Finally it adds instructions in the ``optimal_sizing`` notebook.

### CODE REVIEW

File: temoa-uiuc/data_files/run_mga_scenario03.txt [hunk 1]
```diff
@@ -8,15 +8,15 @@
 # Input File (Mandatory)
 # Input can be a .sqlite or .dat file
 # Both relative path and absolute path are accepted
---input=data_files/03_uiuc.sqlite
+--input=data_files/03_uiuc_mga.sqlite
```
Comment (robfairh) [inline temoa-uiuc/data_files/run_mga_scenario03.txt:11]: should 03_uiuc_mga be already a file in the repo or is this file created with the Snakefile?

File: temoa-uiuc/data_files/run_mga_scenario03.txt [hunk 2]
```diff
@@ -8,15 +8,15 @@
...
 
 # Output File (Mandatory)
 # The output file must be a existing .sqlite file
---output=data_files/03_uiuc.sqlite
+--output=data_files/03_uiuc_mga.sqlite
 
 # Scenario Name (Mandatory)
 # This scenario name is used to store results within the output .sqlite file
---scenario=mga_scenario03
+--scenario=scenario03
```
Comment (robfairh) [inline temoa-uiuc/data_files/run_mga_scenario03.txt:19]: Same here. Is scenario03 created by the Snakefile?

File: temoa-uiuc/Snakefile [hunk 3]
```diff
@@ -13,9 +13,9 @@ rule all:
         expand(r"data_files/{db}_{scenario}_model/{scenario}_OutputLog.log",
                zip,
                db=DBS, scenario=SCENARIOS),
-        expand(r"data_files/{db}_{scenario}_model/{scenario}.lp",
-               zip,
-               db=DBS, scenario=SCENARIOS),
+        # expand(r"data_files/{db}_{scenario}_model/{scenario}.lp",
+        #        zip,
+        #        db=DBS, scenario=SCENARIOS),
         expand("figures/{scenario}_{sector}_capacity.png", sector=SECTORS,
```
Comment (robfairh) [inline temoa-uiuc/Snakefile:19]: does this require the pre-existence of the folder 'figures' inside temoa-uiuc/ or do those lines create the folder too?

**Comment by robfairh** [comment]
This is how your figure looks currently (this is in the temoa_processing notebook). Although the xlabels are readable, they could look better. Try something to make them look better. Maybe make the bars wider, or rotate the labels slightly.

image

**Comment by robfairh** [comment]
The same comment applies for all the new figures in that notebook.

**Comment by samgdotson** [comment]
For the moment, only the figures in ``optimal_sizing`` will be used in the optimal sizing paper.  The wider figures are not going to be in the optimal sizing paper, for now. I can delete these figures but I think it might be better to create a separate issue. What do you think?

**Comment by robfairh** [comment]
This figure is in the notebook 'optimal_sizing':

image

And this figure is from your ANS Anual meeting presentation:

image

I am assuming they solve the same scenario. They are very similar but not identical.
Did you change anything else besides automatizing the process?

**Review by robfairh — COMMENTED** [review]
I made a few comments.
Looks good overall!

**Comment by robfairh** [comment]
> For the moment, only the figures in `optimal_sizing` will be used in the optimal sizing paper. The wider figures are not going to be in the optimal sizing paper, for now. I can delete these figures but I think it might be better to create a separate issue. What do you think?

Let's do that.

File: temoa-uiuc/data_files/run_mga_scenario03.txt (hunk 1 above)
Comment (samgdotson) [reply temoa-uiuc/data_files/run_mga_scenario03.txt:11]: Snakemake creates the required files if they don't already exist by the rule "make_temoa_input"

File: temoa-uiuc/data_files/run_mga_scenario03.txt (hunk 2 above)
Comment (samgdotson) [reply temoa-uiuc/data_files/run_mga_scenario03.txt:19]: It is created by the Snakefile

File: temoa-uiuc/Snakefile (hunk 3 above)
Comment (samgdotson) [reply temoa-uiuc/Snakefile:19]: The folder is also created by the snakefile! The rule "all" tells snakemake what to look for in the outputs and if a required folder doesn't exist it is created.

**Comment by samgdotson** [comment]
> This figure is in the notebook 'optimal_sizing':
> 
> image
> 
> And this figure is from your ANS Anual meeting presentation:
> 
> image
> 
> I am assuming they solve the same scenario. They are very similar but not identical.
> Did you change anything else besides automatizing the process?

I added a slight capital cost to wind power since the presentation, they are otherwise unchanged.

**Comment by samgdotson** [comment]
This PR closes #79 by adding ``optimal_sizing.ipynb`` and closes #84 by updating ``data_parser.py``.

**Comment by robfairh** [comment]
Good! Merging

**Review by robfairh — APPROVED** [review]

**Pull request merged**